packages/react-dom/src/events/accumulateTwoPhaseListeners.js

```diff
@@ -111,18 +111,19 @@ export default function accumulateTwoPhaseListeners(
         if (listeners !== undefined) {
           const captureListeners = Array.from(listeners.captured);
           const bubbleListeners = Array.from(listeners.bubbled);
+          const container = ((lastHostComponent: any): Element);
 
           for (let i = 0; i < captureListeners.length; i++) {
             const listener = captureListeners[i];
             const {callback} = listener;
             dispatchListeners.unshift(callback);
-            dispatchInstances.unshift(((lastHostComponent: any): Element));
+            dispatchInstances.unshift({instance: node, container});
```

Comment:
Can you explain in more detail what's different about Scopes that makes us need this? Why do we need to "remember" the closest "last host component", but we don't for regular events? What is special here? Why can't we figure out the DOM node later?

Reply:
Scopes don't have a physical DOM element representing them as a target, so in order to provide the user with one, we given them the nearest DOM node from the
the native event's original target – this is also how React Flare worked. We can't do this later, as we don't know which part of the tree we came in by – so we have to do this here.

Reply:
Is this about `target` or `currentTarget`?

Reply:
It's about the `currentTarget` on the event object for when events are on scopes. We want it to be the last "real" currentTarget that is in the DOM.